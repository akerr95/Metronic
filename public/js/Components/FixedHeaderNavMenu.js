/**
 * Created by akerr on 8/26/16.
 * Fixed Header nav Menu is always situated at the top of a navigation header.
 */
import React from "react";

var _dropDownMap = new Map();

//TODO Examine what properties should be passed add Proptypes and default types..
export class TopMenuNavigation extends React.Component {
    constructor(props) {
        super(props);
        this._handleChange = this._handleChange.bind(this);
        let len = props.fixedNavigation.iconMapper.length;
        this.state = {};
        this.state = this._createState(len);
    }

    render() {
        let {topMenuNavigation, icon, userProfile} = this.props.classes;
        let stateMap = this._convertArrToMap(this.state.whoIsOpen);
        let icons = this._populateIcons(stateMap, icon);
        return (<div className={topMenuNavigation.divClassName}>
            <ul className={topMenuNavigation.ulClassName}>
                {icons}
                <UserProfile classes={userProfile} updateOpened={this._handleChange} iconState={stateMap}
                             userIcon={this.props.fixedNavigation.userIcon}
                             iconProfiles={this.props.fixedNavigation.iconProfiles}/>
            </ul>
        </div>)
    }

    _populateIcons(stateMap, classes) {
        let icons = [];
        this.props.fixedNavigation.iconMapper.map((icon)=> {
            icons.push(<Icon key={icon.iconName} classes={classes} updateOpened={this._handleChange}
                             iconState={stateMap} {...icon}/>);
        });
        return icons;

    }

    _createState(count) {
        let iconStateParameters = [];
        for (let i = 0; i < count; i++) {
            let names = this.props.fixedNavigation.iconMapper[i].iconName;
            iconStateParameters.push([names, false]);
        }
        let names = this.props.fixedNavigation.userIcon.iconName;
        iconStateParameters.push([names, false]);
        return {whoIsOpen: iconStateParameters};
    }

    _convertArrToMap(iconStateParameters) {
        let stateMap = new Map(iconStateParameters);
        return stateMap;
    }

    _convertMapToArr(stateMap) {
        let iconStateParameters = [];
        for (let [key,value] of stateMap) {
            iconStateParameters.push([key, value]);
        }
        return iconStateParameters;
    }

    _handleChange(state) {
        let iconStateParameters = this._convertMapToArr(state);
        this.setState({whoIsOpen: iconStateParameters});
    }
}

//TODO Examine what properties should be passed add Proptypes and default types..
export class UserProfile extends React.Component {
    constructor(props) {
        super(props);
        this._handleIconClick = IconClicked;
        this._handleIconClick = this._handleIconClick.bind(this);
    }

    render() {
        let {liClassName, userIcon, profileMenu}=this.props.classes;
        let classes = liClassName;
        if (this._isTrue()) {
            classes += " isOpen";
        }

        return (
            <li onClick={this._handleIconClick} className={classes}>
                <UserIcon classes={userIcon} {...this.props.userIcon}/>
                <ProfileMenu classes={profileMenu}
                             updateOpened={this.props.updateOpened}
                             iconState={this.props.iconState}
                             triggerName={this.props.userIcon.iconName}
                             {...this.props.iconProfiles}/>
            </li>);

    }

    _isTrue() {
        let iconName = this.props.userIcon.iconName;
        let stateMap = this.props.iconState;
        return stateMap.get(iconName);
    }

}

//TODO Examine what properties should be passed add Proptypes and default types..
export class Icon extends React.Component {

    constructor(props) {
        super(props);
        this._handleIconClick = IconClicked;
        this._handleIconClick = this._handleIconClick.bind(this);
    }

    render() {
        let {liClassName, aClassName, notify, actionDropDown} = this.props.classes;
        liClassName += this.props.liClassName;
        if (this._isClosedTrue()) {
            liClassName += " isOpen";
        }
        return (<li onClick={this._handleIconClick} className={liClassName}>
            <a className={aClassName}>
                <i className={this.props.iconName}></i>
                <Notice totalNotifications={this.props.notify} {...notify}/>
            </a>
            <ActionDropDown dropDownOpt={this.props.liClassName}
                            actionDropDown={this.props.actionDropDown}
                            triggerName={this.props.iconName}
                            updateOpened={this.props.updateOpened}
                            iconState={this.props.iconState}
                            classes={actionDropDown}/>
        </li>);
    }

    _isClosedTrue() {
        let iconName = this.props.iconName;
        let stateMap = this.props.iconState;
        return stateMap.get(iconName);
    }

}

//TODO Examine what properties should be passed add Proptypes and default types..
export class ProfileMenu extends React.Component {
    constructor(props) {
        super(props);
        this._LostFocus = IconClicked;
        this._LostFocus = this._LostFocus.bind(this);
    }

    render() {
        let {ulClassName, divider, iconProfile} = this.props.classes;
        let iconProfiles = this._PopulateIconProfile(divider, iconProfile);

        return (
            <ul onMouseLeave={this._LostFocus} className={ulClassName}>
                {iconProfiles}
            </ul>
        );
    }

    _PopulateIconProfile(divider, iconClasses) {
        let iconProfiles = [];
        this.props.icons.map((icon, i)=> {
            if (this.props.count !== i) {
                iconProfiles.push(<li key={icon.iconName}><IconProfile classes={iconClasses}{...icon}/></li>);
            } else {
                iconProfiles.push(<Divider classes={divider} key={"Divider"}/>);
                iconProfiles.push(<li key={icon.iconName}><IconProfile classes={iconClasses}{...icon}/></li>);
            }
        });
        return iconProfiles;
    }
}
//TODO Examine what properties should be passed add Proptypes and default types..
export const UserIcon = ({imgLocation, stringName, iconName, classes})=>(
    <a className={classes.aClassName}>
        <img className={classes.imgClassName} src={imgLocation} alt="Image of User"/>
        <span className={classes.spanClassName}>{stringName}</span>
        <i className={iconName}></i>
    </a>
);
UserIcon.propTypes = {
    imgLocation: React.PropTypes.string,
    stringName: React.PropTypes.string.isRequired,
    iconName: React.PropTypes.string.isRequired
};
export const Notice = ({totalNotifications, spanClassName})=>(
    <span className={spanClassName}>{totalNotifications}</span>);
//TODO Examine what properties should be passed add Proptypes and default types..
export const IconProfile = ({iconName, href, stringName, notify, classes})=> {
    let span = <Notice spanClassName={classes.noticeClassName} totalNotifications={notify}/>;
    return (
        <a href={href}>
            <i className={iconName}>{stringName}</i>
            {notify > 0 ? span : false}
        </a>);
};

//TODO Examine what properties should be passed add Proptypes and default types..
// export const ActionDropDownHeader = ({pending,message,classes})=> {
//     let {spanClassName,liClassName } = classes;
//     let newNotifications = <h3><span className={spanClassName}>{pending +" "+ message + " "}</span>notifications</h3>;
//     let noNewNotifications = <h3><span className={spanClassName}>{message}</span></h3>;
//     return (
//
//         <li className={liClassName}>
//             {pending > 0 ? newNotifications : noNewNotifications }
//             <a className={pending > 0 ? "":"hide"} href="#">view all</a>
//         </li>
//     );
// };


/************************
 * State-Full Components*
 ***********************/

// export class ActionDropDown extends React.Component {
//     constructor(props) {
//         super(props);
//         this._LostFocus = IconClicked;
//         this._LostFocus = this._LostFocus.bind(this);
//     }
//
//     render() {
//         let {ulClassName, actionDropDownHeader, slider} =this.props.classes;
//
//         return ( <ul onMouseLeave={()=> {
//             this._LostFocus(this.props.triggerName, this.props.iconState, this.props.updateOpened);
//         }} className={ulClassName}>
//             <ActionDropDownHeader classes={actionDropDownHeader} {...this.props.actionDropDown.headings}/>
//             <li><Slider classes={slider}
//                         dropDownOpt={this.props.dropDownOpt}
//                         messages={this.props.actionDropDown.messages}/>
//             </li>
//         </ul>);
//     }
// }

/***************************
 * Pure Stateless Components*
 ***************************/
export const ActionDropDown = ({data,element, callback, passedState, styles})=>(
    <ul onMouseLeave={function () {
        IconClicked(data.triggerName,
            passedState.iconState,
            callback);
    }} className={styles.ulClassName}>
        {element.header}
        <li>{element.body}</li>
    </ul>
);
ActionDropDown.propTypes = {
    data:React.PropTypes.object,
    element:React.PropTypes.object,
    passedState: React.PropTypes.object.isRequired,
    styles: React.PropTypes.object,
    callback: React.PropTypes.func
};
ActionDropDown.defaultProps = {
    data:{triggerName:"icon-fire"},
    element:{header:<div></div>,body:<div></div>},
    styles:{ulClassName:"dropdown-menu"},
    callback:()=>{console.error("Please specify a callback function")},
};
export const ActionDropDownHeader = ({data, styles})=>(
    <li className={styles.liClassName}>
        <h3 className={styles.head}>
            <span className={styles.bold}>{data.pending + " " + data.subject + " "}</span>
            {data.heading}
        </h3>
        <a className={styles.aClassName}>{data.actionMessage}</a>
    </li>
);
ActionDropDownHeader.propTypes = {
    data: React.PropTypes.object,
    styles: React.PropTypes.object
};
ActionDropDownHeader.defaultProps = {
    data: {pending: "one", subject: "Default", heading: "view all"},
    styles: {head: "", bold: "bold", liClassName: "external", aClassName: ""}
};
export const Divider = ({styles})=>(
    <li className={styles.liClassName}></li>
);
Divider.propTypes = {
    styles: React.PropTypes.object
};
Divider.defaultProps = {
    styles: {liClassName: "divider"}
};
export const ActionDropDownError = ({data, styles})=> (
    <span>
                <span className={styles.err}>{data.errorMessage}</span>
                <ul><Divider/></ul>
                <span className={styles.solution}>{data.solution}</span>
        </span>
);
ActionDropDownError.propTypes = {
    data: React.PropTypes.shape({
        errorMessage: React.PropTypes.string.isRequired,
        solution: React.PropTypes.string.isRequired
    }),
    styles: React.PropTypes.object
};
ActionDropDownError.defaultProps = {
    styles: {err: "error", solution: "solution"}
};
export const Slider = ({data, styles})=> (
    <div className={styles.slider}>
        <ul className={styles.ul}>
            {data}
        </ul>
        <div className={styles.bar}></div>
        <div className={styles.rail}></div>
    </div>
);
Slider.propTypes = {
    data: React.PropTypes.element.isRequired,
    styles: React.PropTypes.object,
};
Slider.defaultProps = {
    styles: {
        slider: "slimScrollDiv",
        ul: "dropdown-menu-list",
        bar: "slimScrollBar",
        rail: "slimScrollRail"
    }
};
export const ClickableList = ({data, styles}) =>(
    <li className={styles.liClassName}>
        <a className={styles.aClassName}>
            {data}
        </a>
    </li>
);
ClickableList.propTypes = {
    data: React.PropTypes.element.isRequired,
    styles: React.PropTypes.object
};
ClickableList.defaultProps = {
    styles: {liClassName: "", aClassName: ""}
};
export const ActionDropDownTask = ({data, styles})=>(
    <span>
       <span className={styles.task}>
           <span className={styles.desc}>{data.desc}</span>
           <span className={styles.percent}>{data.percent + '%'}</span>
       </span>
       <span className={styles.progress}>
           <span className={styles.progressbar}>
               <span className={styles.sr}></span>
           </span>
       </span>
   </span>
);
ActionDropDownTask.propTypes = {
    data: React.PropTypes.object,
    styles: React.PropTypes.object
};
ActionDropDownTask.defaultProps = {
    data: {desc: "Metronic V1.0", percent: 40},
    styles: {
        task: "task",
        desc: "desc",
        percent: "percent",
        progress: "progress",
        progressbar: "progress-bar progress-bar-success",
        sr: "sr-only"
    },
};
export const ActionDropDownInbox = ({data, styles})=>(
    <span>
        <span className={styles.photo}><img src={data.imgLocation} className={styles.img || "img-circle"}/></span>
        <span className={styles.subject}>
            <span className={styles.time}>{data.time}</span>
            <span className={styles.from}>{data.sender}</span>
        </span>
        <span className={styles.message}>{data.message}</span>
    </span>
);
ActionDropDownInbox.propTypes = {
    data: React.PropTypes.object,
    styles: React.PropTypes.object
};
ActionDropDownInbox.defaultProps = {
    data: {
        imgLocation: "https://d13yacurqjgara.cloudfront.net/users/559591/screenshots/2928328/wild_west_bandit_tubik_illustration_art_design_colors_character_brush_adobe_arthur_avakyan_1x.jpg",
        sender: "Alec",
        time: "just now",
        message: "Default message.",
    },
    styles: {
        photo: "photo",
        img: "img-circle",
        subject: "subject",
        time: "time",
        from: "from",
        message: "message"
    }
};
export const ActionDropDownNotify = ({data, styles})=>(
    <span>
            <span className={styles.time}>{data.time}</span>
            <span className={styles.details}>
                <span className={styles.label}>
                    <i className={data.iconName}></i>
                </span>
                {data.message}
            </span>
     </span>
);
ActionDropDownNotify.propTypes = {
    data: React.PropTypes.object,
    styles: React.PropTypes.object
};
ActionDropDownNotify.defaultProps = {
    data: {time: "just now", message: "this is a default message", iconName: "fa fa-plus"},
    styles: {time: "time", details: "details", label: "label label-sm label-icon label-success edited-label"}
};

/*****************
 * Pure Functions*
 *****************/
export function IconClicked(triggerName, iconState, fn) {
    let stateMap = new Map();
    for (let [key, value] of iconState) {
        key === triggerName ? stateMap.set(key, !value) : stateMap.set(key, false);
    }
    fn(stateMap);
}