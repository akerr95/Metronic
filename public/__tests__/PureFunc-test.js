/**
 * Created by akerr on 9/12/16.
 */
import * as PureFunc from '../js/Components/PureFunctions.js';
import * as FixedHeader from "../js/Components/FixedHeaderNavMenu.js";
describe("Pure functions suite test", ()=> {
    function stateCreate(triggerName) {
        let state = new Map();
        for (let i = 0; i < 4; i += 1) {
            i === 0 ? state.set(triggerName, false) : state.set(triggerName + i, false);
        }
        return state;
    }

    it("IconTriggered function", ()=> {
        var triggerName = "icon-test";
        let state = stateCreate(triggerName);
        let form = false;

        const callBack = x => state = x;
        PureFunc.IconTriggered(triggerName, state, callBack);
        expect(state.get(triggerName)).not.toEqual(form);
    });
    describe("CreateBooleanState",()=>{
        let states = ["Test","Test1","Test1","Test2"];

        it("Returns an array containing an array of size 2",()=>{
            let result =PureFunc.CreateBooleanState(states);
            expect(Array.isArray(result.elementBool)).toBeTruthy();
            expect(Array.isArray(result.elementBool[0])).toBeTruthy();
            expect(result.elementBool[0].length).toEqual(2);
        });
        it("Returns array containing no repeats",()=>{
            let result = PureFunc.CreateBooleanState(states).elementBool;
            let myArr = [["Test",false],["Test1",false],["Test2",false]];
            expect(result).toEqual(myArr);
        });
    });
    describe("ConvertArrToMap", ()=> {
        let arr = [["test", false]];
        let map = new Map([["test", false]]);
        it("return map object", ()=> {
            let result = PureFunc.ConvertArrToMap(arr);
            expect(typeof result).toEqual("object");
        });
        it("return correct map object", ()=> {
            let result = PureFunc.ConvertArrToMap(arr);
            expect(result).toEqual(map);
        });
    });
    describe("ConvertMapToArr", ()=> {
        let arr = [["test", false]];
        let map = new Map([["test", false]]);
        it("return arr object", ()=> {
            let result = PureFunc.ConvertMapToArr(map);
            expect(typeof result).toEqual("object");
        });
        it("return correct map object", ()=> {
            let result = PureFunc.ConvertMapToArr(map);
            expect(result).toEqual(arr);
        });
    });
    describe("populateElement", ()=> {
        function createData(amount) {
            let myMap = new Map();

            for (let i = 0; i < amount; i += 1) {
                let data ={key:0,notifyCount:2};
                data.key = i;
                myMap.set({data},FixedHeader.Notice);
            }
            return myMap;
        }


        it("returns a 4 elements for 4 data types", ()=> {
            let amnt = 4;
            let result = PureFunc.populateElement(createData(amnt));
            expect(result.length).toEqual(amnt);
            expect(result[0].hasOwnProperty("type")).toBe(true);

        });
    });
    describe("isOpen", ()=> {

        it("returns isClosed based on state", ()=> {
            let state = stateCreate("test");
            let element = "test";
            let ans = "isClosed";
            let result = PureFunc.isOpen(element, state);
            expect(result).toBe(ans);
        });
        it("returns isOpen based on state", ()=> {
            let state = stateCreate("test");
            let element = "test";
            let ans = "isOpen";
            const changeState = (x)=>(state = x);
            PureFunc.IconTriggered(element, state, changeState);
            let result = PureFunc.isOpen(element, state);
            expect(result).toBe(ans);
        });
        it("returns isCustom based on state", ()=> {
            let state = stateCreate("test");
            let element = "test";
            let ans = "isCustom";
            let custom = {second: ans};

            let result = PureFunc.isOpen(element, state, custom);
            expect(result).toBe(ans);
        });
        it("returns isAlsoCustom based on state", ()=> {
            let state = stateCreate("test");
            let element = "test";
            let ans = "isAlsoCustom";
            let custom = {first: ans};

            const changeState = (x)=>(state = x);
            PureFunc.IconTriggered(element, state, changeState);
            let result = PureFunc.isOpen(element, state, custom);
            expect(result).toBe(ans);
        });
    });
    describe("TopMenuData", ()=> {

        let topMenuData = new PureFunc.TopMenuData();
        it("returns new object data", ()=> {

            expect(topMenuData.constructor).toMatchSnapshot();
            expect(topMenuData).toBeDefined();
        });
        it("added task elements to topMenuData ", ()=> {
            let result = topMenuData.addTasks([
                {desc: "Metronic v1.1", percent: "10"}
                ]);
            expect(result).toBeDefined();
            expect(result.tasks).toMatchSnapshot();
        });
        it("added inbox elements to topMenuData", ()=> {
            let result = topMenuData.addInbox([
                {
                imgLocation: "http://bit.ly/2cfaQ21",
                time: "just now",
                sender: "Alec kerr",
                message: "This was made from a test file."
            }
            ]);
            expect(result).toBeDefined();
            expect(result.inbox).toMatchSnapshot();
        });
        it("added notification elements to topMenuData", ()=> {
            let result = topMenuData.addNotifications([{
                time: "1 min",
                iconName: "icon-plus",
                message: "This was made from a test file.",
            }]);
            expect(result).toBeDefined();
            expect(result.notifications).toMatchSnapshot();
        });
        it("added header element to topMenuData", ()=> {
            let result = topMenuData.addHeader([
                {
                key: "tasks",
                pending: 2,
                subject: "Test",
                heading: "Notifications",
                actionMessage: "view all"
            },{
                key: "inbox",
                pending: 2,
                subject: "Test",
                heading: "Notifications",
                actionMessage: "view all"
            },{
                key: "notifications",
                pending: 2,
                subject: "Test",
                heading: "Notifications",
                actionMessage: "view all"
            }]);
            expect(result).toBeDefined();
            expect(result.header).toMatchSnapshot();
        });
        it("returns correct ActionDropDown",()=>{
            let header = topMenuData.getHeader("notifications");
            let body = topMenuData.getInbox();
            let result = topMenuData.makeActionDropDown(header,body);

            expect(result).toMatchSnapshot();
        });
        it("added Icon element to topMenuData", ()=> {

            let result = topMenuData.addIcon([{
                iconName: "icon-bell",
                key:"tasks",
                notice:{},
                actionDropDown: {},
                classStyles: {
                    liClassName: "dropdown dropdown-extended nav-dropdown dropdown-task ",
                    aClassName: "dropdown-toggle"
                }
            },{
                iconName: "icon-envelope-open",
                key:"inbox",
                notice:{},
                actionDropDown: {},
                classStyles: {
                    liClassName: "dropdown dropdown-extended nav-dropdown dropdown-inbox ",
                    aClassName: "dropdown-toggle"
                }
            },{
                iconName: "icon-calendar",
                key:"notifications",
                notice:{},
                actionDropDown: {},
                classStyles: {
                    liClassName: "dropdown dropdown-extended nav-dropdown dropdown-notification",
                    aClassName: "dropdown-toggle"
                }
            }]);
            expect(result).toBeDefined();
            expect(result.icons).toMatchSnapshot();
            expect(result.stateKeys.length).toEqual(3);
        });
        it("added UserIcon element to topMenuData", ()=> {
            let result = topMenuData.addUserIcon({
                imgLocation: "http://bit.ly/2cqW9ue",
                stringName: "Alec Kerr",
                iconName: "icon-options-vertical"
            });
            expect(result).toBeDefined();
            expect(result.userIcon).toMatchSnapshot();
        });
        it("added IconProfile element to topMenuData", ()=> {
            let result = topMenuData.addIconProfile([{
                iconName: "icon-test",
                stringName: "My Profile",
                href: "#"
            }
            ]);
            expect(result).toBeDefined();
            expect(result.iconProfiles).toMatchSnapshot();
        });
        it("added IconProfile element with divider to topMenuData", ()=> {
            let result = topMenuData.addIconProfile([{
                iconName: "icon-test",
                stringName: "My Profile",
                href: "#"
            },{divider:true}
            ]);
            expect(result).toBeDefined();
            expect(result.iconProfiles).toMatchSnapshot();
        });
        it("added ProfileMenu element to topMenuData", ()=> {
            let result = topMenuData.hasProfileMenu(true);
            expect(result).toBeDefined();
            expect(result.profileMenu).toBeTruthy();
        });
        it("created correct icons",()=>{
            let result = topMenuData._makeIcons();
            expect(result).toMatchSnapshot();
        });
        it("added UserProfile element to topMenuData", ()=> {
            let result = topMenuData.hasUserProfile(true);
            expect(result).toBeDefined();
            expect(result.userProfile).toBeTruthy();
        });
        it("generates TopMenuNav", ()=> {
            let result = topMenuData.generate();
            expect(typeof result).toEqual("object")
        });
    });
});