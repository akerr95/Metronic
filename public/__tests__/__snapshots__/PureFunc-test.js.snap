exports[`Pure functions suite test TopMenuData added Icon element to topMenuData 1`] = `
Array [
  Array [
    "tasks",
    Object {
      "data": Object {
        "actionDropDown": Object {},
        "classStyles": Object {
          "aClassName": "dropdown-toggle",
          "liClassName": "dropdown dropdown-extended nav-dropdown dropdown-task "
        },
        "iconName": "icon-bell",
        "key": "tasks",
        "notice": Object {}
      }
    }
  ],
  Array [
    "inbox",
    Object {
      "data": Object {
        "actionDropDown": Object {},
        "classStyles": Object {
          "aClassName": "dropdown-toggle",
          "liClassName": "dropdown dropdown-extended nav-dropdown dropdown-inbox "
        },
        "iconName": "icon-envelope-open",
        "key": "inbox",
        "notice": Object {}
      }
    }
  ],
  Array [
    "notifications",
    Object {
      "data": Object {
        "actionDropDown": Object {},
        "classStyles": Object {
          "aClassName": "dropdown-toggle",
          "liClassName": "dropdown dropdown-extended nav-dropdown dropdown-notification"
        },
        "iconName": "icon-calendar",
        "key": "notifications",
        "notice": Object {}
      }
    }
  ]
]
`;

exports[`Pure functions suite test TopMenuData added IconProfile element to topMenuData 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "href": "#",
        "iconName": "icon-test",
        "key": "icon-test",
        "stringName": "My Profile"
      }
    },
    [Function IconProfile]
  ]
]
`;

exports[`Pure functions suite test TopMenuData added IconProfile element with divider to topMenuData 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "href": "#",
        "iconName": "icon-test",
        "key": "icon-test",
        "stringName": "My Profile"
      }
    },
    [Function IconProfile]
  ],
  Array [
    Object {
      "data": Object {
        "href": "#",
        "iconName": "icon-test",
        "key": "icon-test",
        "stringName": "My Profile"
      }
    },
    [Function IconProfile]
  ],
  Array [
    Object {
      "data": Object {
        "divider": true,
        "key": "divider1"
      }
    },
    [Function Divider]
  ]
]
`;

exports[`Pure functions suite test TopMenuData added UserIcon element to topMenuData 1`] = `
Object {
  "iconName": "icon-options-vertical",
  "imgLocation": "http://bit.ly/2cqW9ue",
  "stringName": "Alec Kerr"
}
`;

exports[`Pure functions suite test TopMenuData added header element to topMenuData 1`] = `
Array [
  Array [
    "tasks",
    Object {
      "data": Object {
        "actionMessage": "view all",
        "heading": "Notifications",
        "key": "tasks",
        "pending": 2,
        "subject": "Test"
      },
      "fn": [Function ActionDropDownHeader]
    }
  ],
  Array [
    "inbox",
    Object {
      "data": Object {
        "actionMessage": "view all",
        "heading": "Notifications",
        "key": "inbox",
        "pending": 2,
        "subject": "Test"
      },
      "fn": [Function ActionDropDownHeader]
    }
  ],
  Array [
    "notifications",
    Object {
      "data": Object {
        "actionMessage": "view all",
        "heading": "Notifications",
        "key": "notifications",
        "pending": 2,
        "subject": "Test"
      },
      "fn": [Function ActionDropDownHeader]
    }
  ]
]
`;

exports[`Pure functions suite test TopMenuData added inbox elements to topMenuData 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "imgLocation": "http://bit.ly/2cfaQ21",
        "key": "Alec kerr",
        "message": "This was made from a test file.",
        "sender": "Alec kerr",
        "time": "just now"
      }
    },
    [Function ActionDropDownInbox]
  ]
]
`;

exports[`Pure functions suite test TopMenuData added notification elements to topMenuData 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "iconName": "icon-plus",
        "key": "1icon-plus",
        "message": "This was made from a test file.",
        "time": "1 min"
      }
    },
    [Function ActionDropDownNotify]
  ]
]
`;

exports[`Pure functions suite test TopMenuData added task elements to topMenuData  1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "desc": "Metronic v1.1",
        "key": "Metronic v1.1",
        "percent": "10"
      }
    },
    [Function ActionDropDownTask]
  ]
]
`;

exports[`Pure functions suite test TopMenuData created correct icons 1`] = `
Array [
  Array [
    Object {
      "classStyles": Object {
        "aClassName": "dropdown-toggle",
        "liClassName": "dropdown dropdown-extended nav-dropdown dropdown-task "
      },
      "data": Object {
        "actionDropDown": Object {
          "body": Array [
            Array [
              Object {
                "data": Object {
                  "desc": "Metronic v1.1",
                  "key": "Metronic v1.1",
                  "percent": "10"
                }
              },
              [Function ActionDropDownTask]
            ]
          ],
          "header": Array [
            Array [
              Object {
                "data": Object {
                  "actionMessage": "view all",
                  "heading": "Notifications",
                  "key": "tasks",
                  "pending": 1,
                  "subject": "Test"
                }
              },
              [Function ActionDropDownHeader]
            ]
          ]
        },
        "iconName": "icon-bell",
        "key": "tasks",
        "notice": Object {
          "notifyCount": 1
        }
      }
    },
    [Function Icon]
  ],
  Array [
    Object {
      "classStyles": Object {
        "aClassName": "dropdown-toggle",
        "liClassName": "dropdown dropdown-extended nav-dropdown dropdown-inbox "
      },
      "data": Object {
        "actionDropDown": Object {
          "body": Array [
            Array [
              Object {
                "data": Object {
                  "imgLocation": "http://bit.ly/2cfaQ21",
                  "key": "Alec kerr",
                  "message": "This was made from a test file.",
                  "sender": "Alec kerr",
                  "time": "just now"
                }
              },
              [Function ActionDropDownInbox]
            ]
          ],
          "header": Array [
            Array [
              Object {
                "data": Object {
                  "actionMessage": "view all",
                  "heading": "Notifications",
                  "key": "inbox",
                  "pending": 1,
                  "subject": "Test"
                }
              },
              [Function ActionDropDownHeader]
            ]
          ]
        },
        "iconName": "icon-envelope-open",
        "key": "inbox",
        "notice": Object {
          "notifyCount": 1
        }
      }
    },
    [Function Icon]
  ],
  Array [
    Object {
      "classStyles": Object {
        "aClassName": "dropdown-toggle",
        "liClassName": "dropdown dropdown-extended nav-dropdown dropdown-notification"
      },
      "data": Object {
        "actionDropDown": Object {
          "body": Array [
            Array [
              Object {
                "data": Object {
                  "iconName": "icon-plus",
                  "key": "1icon-plus",
                  "message": "This was made from a test file.",
                  "time": "1 min"
                }
              },
              [Function ActionDropDownNotify]
            ]
          ],
          "header": Array [
            Array [
              Object {
                "data": Object {
                  "actionMessage": "view all",
                  "heading": "Notifications",
                  "key": "notifications",
                  "pending": 1,
                  "subject": "Test"
                }
              },
              [Function ActionDropDownHeader]
            ]
          ]
        },
        "iconName": "icon-calendar",
        "key": "notifications",
        "notice": Object {
          "notifyCount": 1
        }
      }
    },
    [Function Icon]
  ]
]
`;

exports[`Pure functions suite test TopMenuData returns correct ActionDropDown 1`] = `
Object {
  "body": Array [
    Array [
      Object {
        "data": Object {
          "imgLocation": "http://bit.ly/2cfaQ21",
          "key": "Alec kerr",
          "message": "This was made from a test file.",
          "sender": "Alec kerr",
          "time": "just now"
        }
      },
      [Function ActionDropDownInbox]
    ]
  ],
  "header": Array [
    Array [
      Object {
        "data": Object {
          "actionMessage": "view all",
          "heading": "Notifications",
          "key": "notifications",
          "pending": 1,
          "subject": "Test"
        }
      },
      [Function ActionDropDownHeader]
    ]
  ]
}
`;

exports[`Pure functions suite test TopMenuData returns new object data 1`] = `[Function TopMenuData]`;
